staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -92,7 +93,8 @@ type crdHandler struct {
 	// atomic.Value has a very good read performance compared to sync.RWMutex
 	// see https://gist.github.com/dim/152e6bf80e1384ea72e17ac717a5000a
 	// which is suited for most read and rarely write cases
-	customStorage atomic.Value
+	customStorage         atomic.Value
+	previousCustomStorage sync.Map
```

Comment:
Spot to expand the *list* of all waitGroups that need to be completed for all existing mutation done.  Remember that it must be a list since you can go through many of these.  Ones that are complete when removing an existing storage don't need to be copied.